### ISSUE TITLE
byte order mark in mzml output

### ISSUE THREAD

**Issue by bernt-matthias** [compomics/ThermoRawFileParser#109]
I'm using OpenMS FileConverter (which uses ThermoRawFileParser 1.2.3) to convert a raw file to mzml. 

The output contains a byte order mark  in the beginning of the file. If I look at the file with less the beginning looks like:

```
<U+FEFF><?xml version="1.0" encoding="utf-8"?>
<indexedmzML xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://psi.hupo.org/ms/mzml http://psidev.info/files/ms/mzML/xsd/mzML1.1.0.xsd" xmlns="http://psi.hupo.org/ms/mzml">
```

This makes downstream tools (e.g. SearchGUI) fail. Can you explain this behaviour or suggest a fix/workaround?

**Comment by bernt-matthias** [comment]
Its the same for mzXML output. Which might suggest that the problem is an underlying library?

**Comment by nielshulstaert** [comment]
Hi,

Did you try the conversion with the standalone ThermoRawFileParser? On Windows, you don't need to install anything else, on Linux you need mono. If you can share the RAW file, I can try it as well.

**Comment by bernt-matthias** [comment]
The file is here: http://139.18.2.180/~maze/180719_NJ_Bastida_TA456_raw.raw
And the used conda environment over here: http://139.18.2.180/~maze/env.json

I was now able to reproduce the problem with:

`ThermoRawFileParser.sh  -i in/180719_NJ_Bastida_TA456_raw.raw -b trp.mzML -f=2 -e`

which is the call used in OpenMS

But not with

`ThermoRawFileParser.sh  -i in/180719_NJ_Bastida_TA456_raw.raw -b trp.mzML  -format 1`

**Comment by bernt-matthias** [comment]
Please tell me if you got the file then I can remove it again.

**Comment by nielshulstaert** [comment]
Yes I received it, I'll try it out and get back to you.

**Comment by nielshulstaert** [comment]
I can't seem to reproduce it with the standalone version or the bioconda one. I think indeed the problem lies somewhere else.

**Comment by bernt-matthias** [comment]
Did you use 1.2.3?

**Comment by bernt-matthias** [comment]
Which viewer do you use for looking at the file? I only see it with `less`, `cat` for instance is not showing it.

**Comment by bernt-matthias** [comment]
Can you execute `file trp.mzML`

For me it tells me `trp.mzML: XML 1.0 document, UTF-8 Unicode (with BOM) text, with very long lines`

Maybe (wild guess) its the cryptostream used here https://github.com/compomics/ThermoRawFileParser/blob/6ce17b66575dd3d4257b04dab96108ce2abb15f3/Writer/MzMlSpectrumWriter.cs#L89 is

**Comment by nielshulstaert** [comment]
I'm using the latest version of the TRFP. I can't see the BOM whit `less` but the `file` command output is the same as yours and the non-indexed mzML just says `XML 1.0 document, ASCII text, with very long lines`. You might be right about the cryptostream, I'll have a look around for a possible fix. Any suggestions are welcome.

**Comment by bernt-matthias** [comment]
Did some research: According to https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmlwriter.create?view=net-5.0 the BOM should be omitted if you use: `Encoding.UTF8()` instead of `Encoding.UTF8` here: https://github.com/compomics/ThermoRawFileParser/blob/6ce17b66575dd3d4257b04dab96108ce2abb15f3/Writer/MzMlSpectrumWriter.cs#L84

See also the remarks over here: https://docs.microsoft.com/en-us/dotnet/api/system.text.utf8encoding?view=net-5.0

**Comment by nielshulstaert** [comment]
That seems to work, thanks! I'll include in the next release.

**Comment by bernt-matthias** [comment]
Excellent. Thanks a lot.

**Comment by nielshulstaert** [comment]
This is fixed in release 1.3.3
